### PR TITLE
fix(configure): update usage errors

### PR DESCRIPTION
**Pull request by francoischalifour**
This updates the usage errors for `configure`.

### CODE REVIEW

File: src/connectors/configure/connectConfigure.js [hunk 1]
```diff
@@ -45,7 +30,11 @@ Full documentation available at https://community.algolia.com/instantsearch.js/v
 export default function connectConfigure(renderFn = noop, unmountFn = noop) {
   return (widgetParams = {}) => {
     if (!isPlainObject(widgetParams.searchParameters)) {
-      throw new Error(usage);
+      throw new Error(
+        withUsage(
+          `The \`searchParameters\` option must be an object (got type "${typeof widgetParams.searchParameters}")`
```
Comment (tkrugg) [inline src/connectors/configure/connectConfigure.js]: There is a gotcha
`.isPlainObject(null) === false` but `typeof null === object`.
In this case the error message will not make sense. May be best to remove the typeof part

File: src/widgets/configure/__tests__/configure-test.js [hunk 2]
```diff
@@ -2,10 +2,20 @@ import configure from '../configure';
...
   it('throws when you pass it a non-plain object', () => {
-    [
-      () => configure(new Date()),
-      () => configure(() => {}),
-      () => configure(/ok/),
-    ].forEach(widget => expect(widget).toThrow(/Usage/));
+    expect(() => {
+      configure(new Date());
+    }).toThrowErrorMatchingInlineSnapshot(`
+"The \`searchParameters\` option must be an object (got type \\"object\\")
```
Comment (tkrugg) [inline src/widgets/configure/__tests__/configure-test.js]: gotcha  😄

Comment (francoischalifour) [reply src/widgets/configure/__tests__/configure-test.js]: Haha good catch!

**Comment by francoischalifour** [comment]
@Haroenv Displaying the value is not trivial because it depends on its type. Feel free to improve the error messages though!

**Pull request merged**